android/src/main/java/com/swmansion/reanimated/layoutReanimation/SharedTransitionManager.java

```diff
@@ -0,0 +1,400 @@
+package com.swmansion.reanimated.layoutReanimation;
+
+import android.app.Activity;
+import android.util.Log;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.ViewParent;
+import com.facebook.react.bridge.ReactContext;
+import com.facebook.react.uimanager.IllegalViewOperationException;
+import com.facebook.react.uimanager.PixelUtil;
+import com.facebook.react.uimanager.ViewGroupManager;
+import com.facebook.react.views.view.ReactViewGroup;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+public class SharedTransitionManager {
+  private final AnimationsManager mAnimationsManager;
+  private NativeMethodsHolder mNativeMethodsHolder;
+  private final List<View> mAddedSharedViews = new ArrayList<>();
+  private final Map<Integer, View> mSharedTransitionParent = new HashMap<>();
+  private final Map<Integer, Integer> mSharedTransitionInParentIndex = new HashMap<>();
+  private boolean mIsSharedTransitionActive;
+  private final Map<Integer, Snapshot> mSnapshotRegistry = new HashMap<>();
+  private final Map<Integer, View> mCurrentSharedTransitionViews = new HashMap<>();
+  private View mTransitionContainer;
+  private final List<View> mRemovedSharedViews = new ArrayList<>();
+
+  public SharedTransitionManager(AnimationsManager animationsManager) {
+    mAnimationsManager = animationsManager;
+  }
+
+  protected void notifyAboutNewView(View view) {
+    mAddedSharedViews.add(view);
+  }
+
+  protected void notifyAboutRemovedView(View view) {
+    mRemovedSharedViews.add(view);
+  }
+
+  @Nullable
+  protected View getTransitioningView(int tag) {
+    return mCurrentSharedTransitionViews.get(tag);
+  }
+
+  protected void viewsDidLayout() {
+    setupSharedTransitionForViews(mAddedSharedViews, true);
+    mAddedSharedViews.clear();
+  }
+
+  protected void onViewsRemoving(int[] tagsToDelete) {
```

Comment:
```suggestion
  protected void onViewsRemoval(int[] tagsToDelete) {
```
or maybe
```suggestion
  protected void notifyAboutViewsRemoval(int[] tagsToDelete) {
```
I'm not really sure, I just don't like the current name. 😛